src/main/scala/com/mwz/sonar/scala/scoverage/ScoverageReportParser.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Sonar Scala Plugin
+ * Copyright (C) 2018 All contributors
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package com.mwz.sonar.scala.scoverage
+
+import scala.xml.{Node, NodeSeq, XML}
+
+/**
+ *  Scoverage XML reports parser.
+ *
+ *  @author BalmungSan
+ */
+trait ScoverageReportParser extends ScoverageReportParserAPI {
+
+  /** Parses the scoverage report from a file and returns the [[ModuleCoverage]] */
+  override def parse(scoverageReportFilename: String): ModuleCoverage = {
+    val scoverageXMLReport = XML.loadFile(scoverageReportFilename)
+
+    val moduleScoverage = extractScoverageFromNode(scoverageXMLReport)
+
+    val classCoverages = for {
+      classNode <- scoverageXMLReport \\ "class"
+      filename = s"src/main/scala/${classNode \@ "filename"}"
```

Comment:
I'm worried about the hard coded "src/main/scala/" path.
I don't know if it is completely safe to assume all projects will have this root folder, or if we should add a sonar property for the root folder and default it to "src/main/scala/" .

Reply:
So there is already an existing property called `sonar.sources` (which we should probably also rename to `sonar.scala.sources`), which defaults to [`src/main/scala`](https://github.com/mwz/sonar-scala/blob/160810e3030526ae187b7fc0183937d26ff70b45/src/main/scala/com/buransky/plugins/scoverage/sensor/ScoverageSensor.scala#L63) if not set, however, keep in mind that it should be possible to set this for each module individually like e.g. [here](https://github.com/mwz/sonar-scala/blob/79fa93966df0456147339887be15bdc1f1c669d0/samples/sbt/multi-module/sonar-project.properties). Even though the standard for Scala sources is `src/main/scala`, we should still retain that flexibility which the current version of sonar-scala provides.

Reply:
Completely agree, I will make the change.